### PR TITLE
Better customization of zsh plugins

### LINKED ISSUE #61: Improve management of custom zsh plugins
With https://github.com/TheGrowingPlant/mac-playbook/pull/60, we are now adding 3 externals zsh plugins.

For this 3 plugins, we have 2 different modes of operation:
- plugin is installed but need to be activate using config key
- plugin is installed and always activated

I think we could improve this in order to have only one consistent way for handling custom zsh plugins and at the same time, allow user to add their own plugins through configuration.

```yaml
zsh_custom_plugins:
  - name: symfony
    url: git@github.com:TheGrowingPlant/symfony.plugin.zsh.git
  - name: zsh-autosuggestions
    url: https://github.com/zsh-users/zsh-autosuggestions.git
  - name: zsh-syntax-highlighting
    url: https://github.com/zsh-users/zsh-syntax-highlighting.git
```
Custom plugins should be added at the end of the list and order should be preserved for plugin like zsh-syntax-highlighting which need to be loaded after others.

I identify one problem with this config: we are adding special config to `zshrc` for zsh-syntax-highlighting plugin. Maybe we could handle this using an `extra_config` key or something like this.

### PR DESCRIPTION
**Pull request by notFloran**
Fix #61 #62 

For extra config, the user must use `~/.zshrc.local`.

### CODE REVIEW

**Comment by notFloran** [comment]
I fix my shit

File: roles/dotfiles/tasks/main.yml [hunk 1]
```diff
@@ -5,23 +5,12 @@
...
-    repo: https://github.com/zsh-users/zsh-autosuggestions
-    dest: ~/.oh-my-zsh/custom/plugins/zsh-autosuggestions
-    force: yes
-
-- name: Clone symfony plugin
-  git:
-    repo: https://github.com/TheGrowingPlant/symfony.plugin.zsh.git
-    dest: ~/.oh-my-zsh/custom/plugins/symfony
+    repo: "{{ item.url }}"
+    dest: "~/.oh-my-zsh/custom/plugins/{{ item.name }}"
```
Comment (nclsHart) [inline roles/dotfiles/tasks/main.yml]: I don't know ansible best practices but should we clean the dest dir before cloning plugins?

File: default.config.yml [hunk 2]
```diff
@@ -39,6 +39,23 @@ managed_plists:
...
+  - symfony
+  - zsh-syntax-highlighting
+
+zsh_custom_plugins:
+  - name: symfony
+    url: git@github.com:TheGrowingPlant/symfony.plugin.zsh.git
+  - name: zsh-syntax-highlighting
+    url: https://github.com/zsh-users/zsh-syntax-highlighting.git
+
 zsh_syntax_highlighting: true
```
Comment (nclsHart) [inline default.config.yml]: This 2 lines are not needed  anymore.

File: roles/dotfiles/tasks/main.yml [hunk 3]
```diff
@@ -5,23 +5,25 @@
...
-    force: yes
+- name: Remove old zsh custom plugins
+  file:
+    state: absent
+    path: "{{ item.path }}"
+  when: (item.path | basename) not in zsh_custom_plugins|to_json
+  with_items: "{{ current_installed_zsh_custom_plugins.files }}"
 
-- name: Clone symfony plugin
+- name: Clone custom plugin
```
Comment (nclsHart) [inline roles/dotfiles/tasks/main.yml]:
```suggestion
- name: Clone custom plugins
```

**Pull request merged**